a11ym

```diff
@@ -250,4 +263,43 @@ crawler.on('fetchcomplete', function(queueItem) {
     urls.push({url: url});
 });
 
-crawler.start();
+var computeURLToCrawler = function (crawler) {
+    var first = true;
+
+    return function (value) {
+        var url = crawler.processURL(value);
+        console.log(url);
+
+        if (true === first) {
+            first                   = false;
+            crawler.host            = url.host;
+            crawler.initialProtocol = url.protocol;
+            crawler.initialPort     = url.port;
+            crawler.initialPath     = url.path;
+        } else {
+            ++maximumUrls;
+        }
+
+        crawler.queue.add(
+            url.protocol,
+            url.host,
+            url.port,
+            url.path
+        );
+    };
+};
+
+if (1 === program.args.length && '-' === program.args[0]) {
+    crawler.maxDepth = 1;
+    readline
+        .createInterface(process.stdin, undefined)
+        .on('line', computeURLToCrawler(crawler))
+        .on('close', function () { crawler.start(); });
```

Comment:
In some edge cases, the crawler could compute the URL, then the queue is empty, it stops, and we finally add another URL to the queue, but the crawler has already stopped.

I wonder what happens if we call `crawler.start` more than once…